daemon/images/image_pull.go

```diff
@@ -92,7 +92,7 @@ func (i *ImageService) GetRepository(ctx context.Context, ref reference.Named, a
 	// get repository info
 	repoInfo, err := i.registryService.ResolveRepository(ref)
 	if err != nil {
-		return nil, false, err
+		return nil, false, errdefs.InvalidParameter(err)
```

Comment:
Only error returned comes from https://github.com/moby/moby/blob/8aa694c91ac727351ebb1deaacad62149e984fd6/registry/config.go#L342-L344

Reply:
Maybe this should be specified as invalid param in registry/config.go instead.
Presuming that this is an invalid param from the registry service seems a bit too strong of an assumption.

Reply:
Good suggestion, I'll make that change 👍